GeosCore/tracer_mod.F90

```diff
@@ -0,0 +1,760 @@
+    CHARACTER(LEN=63)      :: OrigUnit
+
+    ! Arrays
+    REAL(fp)               :: Flux(State_Grid%NX,State_Grid%NY)
+    REAL(fp)               :: Mask(State_Grid%NX,State_Grid%NY,State_Grid%NZ)
+
+    ! Pointers to fields in the HEMCO data structure
+    REAL(f4),      POINTER :: O3(:,:,:)   => NULL()
+
+    ! Objects
+    TYPE(Species), POINTER :: SpcInfo
+
+    !========================================================================
+    ! Tracer_Source_Phase begins here!
+    !========================================================================
+
+    ! Initialize
+    RC          = GC_SUCCESS
+    ErrMsg      = ''
+    ThisLoc     = &
+       ' -> at Tracer_Source_Phase (in module GeosCore/tracer_mod.F90)'
+    DT          = GET_TS_CHEM()
+    Local_Tally = 0.0_fp
+    Total_Area  = 0.0_fp
+    Total_Spc   = 0.0_fp
+    Mask        = 1.0_fp
+    Flux        = 0.0_fp
+
+    !=======================================================================
+    ! Convert species units to v/v dry
+    !=======================================================================
+    CALL Convert_Spc_Units( Input_Opt, State_Chm, State_Grid, State_Met, &
+                            'v/v dry', RC, OrigUnit=OrigUnit )
+    IF ( RC /= GC_SUCCESS ) THEN
+       ErrMsg = 'Unit conversion error (kg/kg dry -> v/v dry)'
+       CALL GC_Error( ErrMsg, RC, ThisLoc )
+       RETURN
+    ENDIF
+
+    !========================================================================
+    ! Apply tracer source
+    !========================================================================
+
+    ! Loop over species
+    DO N = 1, State_Chm%nAdvect
+
+       ! Point to the Species Database entry for species N
+       SpcInfo => State_Chm%SpcData(N)%Info
+
+       ! Skip Rn-Pb-Be tracers for now
+       IF ( SpcInfo%Is_RadioNuclide ) CYCLE
+
+       ! Skip if this species does not have a source or if the 
+       ! source is handled by HEMCO
+       IF ( TRIM(SpcInfo%Src_Mode) == 'none'   .or. &
+            TRIM(SpcInfo%Src_Mode) == 'HEMCO' ) CYCLE
+
+       IF ( First ) THEN
+ 
+          !------------------------------------------------------------------
+          ! Convert Src_Value to v/v or to seconds for aoa tracers
+          !------------------------------------------------------------------
+          IF(TRIM(SpcInfo%Src_Units) ==   'ppmv'    ) THEN
+
+             SpcInfo%Src_Value = SpcInfo%Src_Value * 1.0E-06
+
+          ELSE IF(TRIM(SpcInfo%Src_Units) ==   'ppbv'    ) THEN
+
+             SpcInfo%Src_Value = SpcInfo%Src_Value * 1.0E-09
+
+          ELSE IF(TRIM(SpcInfo%Src_Units) ==   'pptv'    ) THEN
+
+             SpcInfo%Src_Value = SpcInfo%Src_Value * 1.0E-12
+
+          ELSE IF ( TRIM(SpcInfo%Src_Units) == 'timestep' ) THEN
+
+             IF ( TRIM(SpcInfo%Units) == 'hours' ) THEN
+
+                SpcInfo%Src_Value = SpcInfo%Src_Value * DT / ( 3600_fp )
+
+             ELSE IF(TRIM(SpcInfo%Units) == 'days'  ) THEN
```

Comment:
I put some comments on other files about using SpcInfo%Units. How about SpcInfo%Lifetime_Units, or something like that, so as not to confuse it with other types of units.